Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+  numericity, that all Left options are less than all Right options. This other condition is
+  proven later in `onag_1_10_ii`.
+-/
+
+lemma invVal_numeric {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (i : InvTy {i // 0 < L i} r b) :
+      (invVal (fun (i : {i // 0 < L i}) => L i) R (fun i => (L i).inv') (fun j => (R j).inv')
+        (PGame.mk l r L R) i).Numeric := by
+  induction i <;> simp only [invVal]
+  · exact numeric_zero
+  all_goals
+    case _ i j ih =>
+      apply Numeric.mul
+      · apply Numeric.add
+        · exact numeric_one
+        · apply Numeric.mul
+          · apply Numeric.sub
+            · first | exact h5.2.1 i | exact h5.2.2 i
+            · exact h5
+          · exact ih
+      · first | exact h4 i | exact h3 i i.2
+
+lemma inv'_numeric (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    (∀ i, (x.inv'.moveLeft i).Numeric) ∧ (∀ j, (x.inv'.moveRight j).Numeric) := by
+  constructor <;> (
+    intro i
+    rcases x with ⟨xl, xr, xL, xR⟩
+    exact invVal_numeric ih_xl ih_xr x_num i
+  )
+
+lemma inv'_numeric_left (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ i, (x.inv'.moveLeft i).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).1
+
+lemma inv'_numeric_right (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ j, (x.inv'.moveRight j).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).2
+
+/-! ### Helper lemmas for the main proof
+
+  These lemmas do the heavy lifting within the proof of the main result `onag_1_10`.
+  They are stated with additional hypotheses which will be satisfied by induction
+  during the main proof.
+
+-/
+
+/-- Given `y''`, return `(x', x'_inv, y')` -/
+def components {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (x_pos : 0 < PGame.mk l r L R)
+    (i'' : InvTy {i // 0 < L i} r b) :=
+  match i'' with
+  | InvTy.zero => (0, 0, 0)
+  | InvTy.left₁ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.left₂ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+  | InvTy.right₁ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.right₂ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+
+/-- The identity `1 - x*y'' = (1 - x * y')*(x' - x)/x'` -/
```

Comment:
This, if perhaps not in this preliminary form, seems like a potentially useful equality later down the line. So I think it makes sense to try and come up with a better name. Maybe `one_neg_div_option`?